File: 1413-minimum-value-to-get-positive-step-by-step-sum/1413-minimum-value-to-get-positive-step-by-step-sum.java
class Solution {
    public int minStartValue(int[] nums) {
        int n = nums.length;
        long sum = 0;
        long minSum = 0;
        for(int x : nums){
            sum += x;
            minSum = Math.min(minSum , sum);
        }
        return 1-(int)minSum;
    }
}

File: 1413-minimum-value-to-get-positive-step-by-step-sum/1413-minimum-value-to-get-positive-step-by-step-sum.cpp
class Solution {
public:
    int minStartValue(vector<int>& nums) {
        int n = nums.size();
        long long sum = 0;
        long long minSum = 0;
        for(int x : nums){
            sum += x;
            minSum = min(minSum , sum);
        }
        return 1-minSum;
    }
};